pkg/runtime/wfengine/workflow.go

```diff
@@ -385,23 +393,33 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 	// will trigger this callback channel.
 	callback := make(chan bool)
 	wi.Properties[CallbackChannelProperty] = callback
+	// Request to execute workflow
 	wfLogger.Debugf("Workflow actor '%s': scheduling workflow execution with instanceId '%s'", actorID, wi.InstanceID)
 	// Schedule the workflow execution by signaling the backend
 	err = wf.scheduler(ctx, wi)
 	if err != nil {
 		if errors.Is(err, context.DeadlineExceeded) {
+			// Workflow execution scheduling request failed with recoverable error, record metrics.
+			diag.DefaultWorkflowMonitoring.ExecutionEvent(ctx, "dapr", diag.Workflow, diag.StatusRetryable, 0)
 			return newRecoverableError(fmt.Errorf("timed-out trying to schedule a workflow execution - this can happen if there are too many in-flight workflows or if the workflow engine isn't running: %w", err))
 		}
 		return newRecoverableError(fmt.Errorf("failed to schedule a workflow execution: %w", err))
 	}
 
 	select {
 	case <-ctx.Done(): // caller is responsible for timeout management
+		// Workflow execution failed with non-recoverable error, record metrics.
+		diag.DefaultWorkflowMonitoring.ExecutionEvent(ctx, "dapr", diag.Workflow, diag.StatusFailed, 0)
 		return ctx.Err()
 	case completed := <-callback:
 		if !completed {
+			// Workflow execution failed with Recoverable Error, record metrics
+			diag.DefaultWorkflowMonitoring.ExecutionEvent(ctx, "dapr", diag.Workflow, diag.StatusRetryable, 0)
+
 			return newRecoverableError(errExecutionAborted)
 		}
+		// workflow execution completed, record metrics
```

Comment:
Better add a comment here: `execution latency for workflow is not supported yet`